### ISSUE TITLE
A syntax error that Merlin does not report

### ISSUE THREAD

**Issue by francoisthire** [ocaml/merlin#750]
On the following example, Merlin does not report a syntax error but probably it should:

```ocaml
type foo = Bar of int * int

let test t =
  let rec test2 t =
    match t with
    | Bar(,_) -> ()
  in
  test2 t
```

**Comment by let-def** [comment]
I get:
```
Syntax error inside `match' after unclosed (, expecting pattern
```
What is happening for you?

**Comment by francoisthire** [comment]
It is an issue related to ocamlmerlin.

There is a bug in version 2.5.3 and OCaml 4.02.3 but it is fixed in version 3.0.3
